src/components/layout/GradientBottomActions.tsx

```diff
@@ -1,18 +1,34 @@
 import * as React from "react";
-import { View, StyleSheet, ViewStyle, StyleProp } from "react-native";
+import { StyleProp, StyleSheet, View, ViewStyle } from "react-native";
 import LinearGradient from "react-native-linear-gradient";
 import Animated from "react-native-reanimated";
+import { IOColors, IOSpacer, IOVisualCostants, hexToRgba } from "../../core";
 import { WithTestID } from "../../utils/types";
-import { IOColors, hexToRgba, IOSpacer, IOVisualCostants } from "../../core";
+import {
+  ButtonLink,
+  ButtonLinkProps,
+  ButtonSolid,
+  ButtonSolidProps
+} from "../buttons";
 import { VSpacer } from "../spacer";
 
+export interface ButtonLinkAction {
+  type: "ButtonLink";
+  actionProps: ButtonLinkProps;
+}
+
+export interface ButtonSolidAction {
+  type: "ButtonSolid";
+  actionProps: ButtonSolidProps;
+}
+
 export type GradientBottomActions = WithTestID<{
   transitionAnimStyle: Animated.AnimateStyle<StyleProp<ViewStyle>>;
   dimensions: GradientBottomActionsDimensions;
   // Accepted components: ButtonSolid, ButtonLink
   // Don't use any components other than this, please.
-  primaryAction?: React.ReactNode;
-  secondaryAction?: React.ReactNode;
+  primaryAction?: ButtonSolidAction;
+  secondaryAction?: ButtonLinkAction;
```

Comment:
A definite improvement over the first iteration. What do you think about assigning the `Button[...]Props` directly? Does it make sense?

```suggestion
  primaryAction?: ButtonSolidProps;
  secondaryAction?: ButtonLinkProps;
```